lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2686,7 +2686,12 @@ class HSADevice final : public KalmarDevice
             case hc::EF_AMDGPU_MACH_AMDGCN_GFX906 : triple.append("906"); break;
         }
 
-        const auto isa{get_isa_name_from_triple(std::move(triple))};
+        unsigned SRAM_ECC_ENABLED = reader.get_flags() & hc::EF_AMDGPU_SRAM_ECC;
+        unsigned XNACK_ENABLED = reader.get_flags() & hc::EF_AMDGPU_XNACK;
+        if (SRAM_ECC_ENABLED) isa.append("+sram-ecc");
+        if (XNACK_ENABLED) isa.append("+xnack");
```

Comment:
Does this even work?  `isa` are used here but only gets defined 2 lines below

Reply:
sorry.  I corrected this in the latest commit